lib/ferrum/screencaster.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+require 'fileutils'
+
+# Combine the resulting frames together into a video with:
+# ffmpeg -y -framerate 30 -i 'frame-%d.jpeg' -c:v libx264 -r 30 -pix_fmt yuv420p try2.mp4
+#
+module Ferrum
+  class Screencaster
+    def initialize(page)
+      @page = page
+      @frame_number = 0
+      @threads = []
+      @base_dir = ""
+    end
+
+    def add_frame(params)
+      @page.command("Page.screencastFrameAck", sessionId: params["sessionId"])
+
+      t = Thread.new { File.binwrite("#{recordings_dir}/frame-#{@frame_number}.jpeg", Base64.decode64(params["data"])) }
+      @frame_number += 1
+      @threads << t
+      true
+    end
+
+    def recordings_dir
+      return @recordings_dir if defined? @recordings_dir
+
+      timestamp = (Time.now.to_f * 1000).to_i
+      @recordings_dir = FileUtils.mkdir_p("#{@base_dir}/screencast_recordings/#{timestamp}/").first
```

Comment:
The directory structure needs cleaned up.
For system/feature specs you probably want to split them up by file then test name.
In other cases you want to split them up by invocation maybe? 

It might be easier to have the caller just pass in an absolute path in #start_screencast rather than a base directory.